### ISSUE TITLE
Issue

### ISSUE THREAD

**Issue by jaday88** [Kara-4search/HellgateLoader_CSharp#1]
![Screenshot_2](https://user-images.githubusercontent.com/95341943/150406725-4e5006e4-d4e6-40f0-b476-23648cc7be84.png)

**Comment by Kara-4search** [comment]
It been updated since 20211204, check the update history in README.md 
r u sure you are using the latest version?

**Comment by jaday88** [comment]
> It been updated since 20211204, check the update history in README.md r u sure you are using the latest version?

Yes, I use the last version 20211204. Your code, which builds by default, gives the same error.

**Comment by Kara-4search** [comment]
> Your

Complied as DEBUG or RELEASE?

**Comment by Kara-4search** [comment]
> > It been updated since 20211204, check the update history in README.md r u sure you are using the latest version?
> 
> Yes, I use the last version 20211204. Your code, which builds by default, gives the same error.


Complied as DEBUG or RELEASE?

**Comment by jaday88** [comment]
> > > It been updated since 20211204, check the update history in README.md r u sure you are using the latest version?
> > 
> > 
> > Yes, I use the last version 20211204. Your code, which builds by default, gives the same error.
> 
> Complied as DEBUG or RELEASE?

Debug x64

**Comment by jaday88** [comment]
> > > > It been updated since 20211204, check the update history in README.md r u sure you are using the latest version?
> > > 
> > > 
> > > Yes, I use the last version 20211204. Your code, which builds by default, gives the same error.
> > 
> > 
> > Complied as DEBUG or RELEASE?
> 
> Debug x64

I insert my shell code, remove the number of bytes (I tried to write the number of bytes), in brackets []. I'm compiling to release x64. A cmd window appears after starting the exe and closes, the shell code does not work. Checked on several shells. I get the code from exe using HxD.

**Comment by Kara-4search** [comment]
I guess the shellcode should not be the problem，try
<img width="896" alt="image" src="https://user-images.githubusercontent.com/85099969/150539563-e8ce9e39-f9b3-464d-a8bf-f569cbc1c54b.png">
Unselect the checkbox in Debug

**Comment by Kara-4search** [comment]
And I update the project today, make sure you are using the lastest version.

**Comment by Kara-4search** [comment]
(After you unselect the checkbox)
When you compile as Debug the first op code should be E9
<img width="809" alt="image" src="https://user-images.githubusercontent.com/85099969/150540599-5b1e8c75-17e3-47eb-9703-88e10ed36a5d.png">

**Comment by jaday88** [comment]
Debug x64
![1](https://user-images.githubusercontent.com/95341943/150547019-f33eb546-9eb6-4601-8c93-bd8998c4c8e7.jpg)
4

**Comment by jaday88** [comment]
Release x6
![2](https://user-images.githubusercontent.com/95341943/150547152-306c9721-48c7-4f3e-9757-86f8624baf3d.jpg)
4

**Comment by jaday88** [comment]
I do everything in the new version. By default, where is the calculator, I rebuilt everything, the calculator only worked for me where x64 Release. The shellcode still doesn't work.

**Comment by Kara-4search** [comment]
> I guess the shellcode should not be the problem，try
> <img alt="image" width="896" src="https://user-images.githubusercontent.com/85099969/150539563-e8ce9e39-f9b3-464d-a8bf-f569cbc1c54b.png">
> Unselect the checkbox in Debug

About DEBUG, Are you sure, you have unselected the checkbox in the picture?
About the shellcode,  test your shellcode in this project https://github.com/Kara-4search/Simple_ShellCodeLoader_CSharp, before you using hellgate.
The API call chain is the same between hellgateloader & https://github.com/Kara-4search/Simple_ShellCodeLoader_CSharp

**Comment by jaday88** [comment]
Yes, unselected the checkbo
![3](https://user-images.githubusercontent.com/95341943/150551397-dde5d996-8799-4f87-b18e-d724786ee9f5.jpg)
x

**Comment by Kara-4search** [comment]
> Yes, unselected the checkbo
> ![3](https://user-images.githubusercontent.com/95341943/150551397-dde5d996-8799-4f87-b18e-d724786ee9f5.jpg)
> x

About DEBUG, If you still got "Cannot generate RWX memory" after unselecting the checkbox , that is kind of weird ~
And from what I see in your pic, RELEASE works fine(the calc could pop up after all)

**Comment by jaday88** [comment]
Simple_ShellCodeLoader_CSharp Doesn't work either. Testing the shellcode DCrat on github. The console window appears for a second and disappears, then nothing happens. Calculator worked on release x64 Hellgate

**Comment by Kara-4search** [comment]
> Simple_ShellCodeLoader_CSharp Doesn't work either. Testing the shellcode DCrat on github. The console window appears for a second and disappears, then nothing happens. Calculator worked on release x64 Hellgate

How do u generate ur shellcode? Could u be more specific

**Comment by jaday88** [comment]
> > Simple_ShellCodeLoader_CSharp Doesn't work either. Testing the shellcode DCrat on github. The console window appears for a second and disappears, then nothing happens. Calculator worked on release x64 Hellgate
> 
> How do u generate ur shellcode? Could u be more specific

I generate in the DCrat builder a shell code in the form of an exe, then I transfer (exe) it to HxD, from there I export the code in the form of a csharp, replace your calculator code with my own, compile it.

**Comment by jaday88** [comment]
![photo_2022-01-21_19-14-27](https://user-images.githubusercontent.com/95341943/150561614-ad522e45-5301-4946-a8b5-cca270f8b2cd.jpg)

**Comment by Kara-4search** [comment]
> > > Simple_ShellCodeLoader_CSharp Doesn't work either. Testing the shellcode DCrat on github. The console window appears for a second and disappears, then nothing happens. Calculator worked on release x64 Hellgate
> > 
> > 
> > How do u generate ur shellcode? Could u be more specific
> 
> I generate in the DCrat builder a shell code in the form of an exe, then I transfer (exe) it to HxD, from there I export the code in the form of a csharp, replace your calculator code with my own, compile it.

I need 2 test that later
